src/test/java/seedu/agendum/logic/LogicManagerTest.java

```diff
@@ -525,6 +525,51 @@ public void execute_find_matchesIfAnyKeywordPresent() throws Exception {
     }
 
 
+    @Test
+    public void execute_undo_identifiesNoPreviousCommand() throws Exception {
+        assertCommandBehavior("undo", UndoCommand.MESSAGE_FAILURE, new ToDoList(), Collections.emptyList());
+    }
+
+    @Test
+    public void execute_undo_reversePreviousMutatingCommand() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
```

Comment:
For now its fine, I think in the future we should separate the test in this file into their own tests for each command. This file is growing to be a little too big right now. Filed #45